File: TH_Buoi1/LinearSearch_Bai1.cpp
#include <iostream>
#include <vector>

int LinearSearch(const std::vector<int>& A, int x) {
    for (size_t i = 0; i < A.size(); ++i) {
        if (A[i] == x) {
            return static_cast<int>(i);
        }
    }
    return -1; // Not found
}

int main() {
    std::vector<int> A = {3, 5, 2, 9, 1};
    int x = 9;
    int index = LinearSearch(A, x);
    if (index != -1) {
        std::cout << "Gia tri " << x << " duoc tim thay tai vi tri: " << index << std::endl;
    } else {
        std::cout << "Gia tri " << x << " khong ton tai trong mang." << std::endl;
    }
    return 0;
}

File: TH_Buoi1/LinearSearch_Bai1.jl
function LinearSearch(A::Vector{Int}, x::Int)
    for i in 1:length(A)
        if A[i] == x
            return i - 1  # To match 0-based indexing if needed
        end
    end
    return -1  # Not found
end

function main()
    A = [3, 5, 2, 9, 1]
    x = 9
    index = LinearSearch(A, x)
    if index != -1
        println("Giá trị $x được tìm thấy tại vị trí: $index")
    else
        println("Giá trị $x không tồn tại trong mảng.")
    end
end

main()